src/sentry/static/sentry/app/components/events/interfaces/spans.tsx

```diff
@@ -0,0 +1,467 @@
+        acc.traceStartTimestamp = span.start_timestamp;
+      }
+
+      // establish trace end timestamp
+
+      const hasEndTimestamp = _.isNumber(span.timestamp);
+
+      if (!acc.traceEndTimestamp) {
+        if (hasEndTimestamp) {
+          acc.traceEndTimestamp = span.timestamp;
+          return acc;
+        }
+
+        acc.traceEndTimestamp = span.start_timestamp;
+        return acc;
+      }
+
+      if (hasEndTimestamp && span.timestamp! > acc.traceEndTimestamp) {
+        acc.traceEndTimestamp = span.timestamp;
+        return acc;
+      }
+
+      if (span.start_timestamp > acc.traceEndTimestamp) {
+        acc.traceEndTimestamp = span.start_timestamp;
+      }
+
+      return acc;
+    }, init);
+
+    // sort span children by their start timestamps in ascending order
+
+    _.forEach(reduced.lookup, spanChildren => {
+      spanChildren.sort((firstSpan, secondSpan) => {
+        if (firstSpan.start_timestamp < secondSpan.start_timestamp) {
+          return -1;
+        }
+
+        if (firstSpan.start_timestamp === secondSpan.start_timestamp) {
+          return 0;
+        }
+
+        return 1;
+      });
+    });
+
+    return reduced;
+  };
+
+  renderSpan = ({
+    spanID,
+    traceID,
+    lookup,
+    span,
+    generateBounds,
+  }: {
+    spanID: string;
+    traceID: string;
+    span: Readonly<SpanType>;
+    lookup: Readonly<LookupType>;
+    generateBounds: (bounds: SpanBoundsType) => SpanGeneratedBoundsType;
+  }) => {
+    const spanChildren: SpanType[] = _.get(lookup, spanID, []);
+
+    const renderedSpanChildren = spanChildren.map(span => {
+      const key = `${traceID}${span.span_id}`;
+
+      // if (
+      //   start_timestamp === undefined ||
+      //   (_.has(span, 'start_timestamp') &&
+      //     _.isNumber(span.start_timestamp) &&
+      //     span.start_timestamp < start_timestamp)
+      // ) {
+      //   start_timestamp = span.start_timestamp;
+      // }
+
+      // if (
+      //   !end_timestamp ||
+      //   (_.has(span, 'timestamp') &&
+      //     _.isNumber(span.timestamp) &&
+      //     span.timestamp > end_timestamp)
+      // ) {
+      //   end_timestamp = span.timestamp;
+      // }
+
+      return (
+        <React.Fragment key={key}>
+          {this.renderSpan({
+            span,
+            spanID: span.span_id,
+            traceID,
+            lookup,
+            generateBounds,
+          })}
+        </React.Fragment>
+      );
+    });
+
+    return (
+      <React.Fragment>
+        <Span span={span} generateBounds={generateBounds} />
+        {renderedSpanChildren}
+      </React.Fragment>
+    );
+  };
+
+  renderRootSpan = () => {
+    const {event} = this.props;
+
+    const trace: TraceType | undefined = _.get(event, 'contexts.trace');
+
+    if (!trace) {
+      return null;
+    }
+
+    const parsedTrace = this.parseTrace();
+
+    // TODO: ideally this should be provided
```

Comment:
Would you want this from the API? The API could ensure that all spans are ordered by start timestamp.

Reply:
Yep :D 

I saw https://github.com/getsentry/sentry/pull/14007